scipy/sparse/linalg/isolve/iterative.py

```diff
@@ -814,3 +825,185 @@ def id(b):
         info = iter_
 
     return postprocess(x), info
+
+
+def _gmres_mgs(A, b, x0=None, restart=None, tol=1e-5, maxiter=None, M=None, callback=None, restrt=None):
+    """
+    Use Restarted Generalized Minimal RESidual based on modified Gram-Schmidt iteration to solve ``Ax = b``.
+
+    Parameters
+    ----------
+    A : {sparse matrix, dense matrix, LinearOperator}
+        The real or complex N-by-N matrix of the linear system.
+        Alternatively, ``A`` can be a linear operator which can
+        produce ``Ax`` using, e.g.,
+        ``scipy.sparse.linalg.LinearOperator``.
+    b : {array, matrix}
+        Right hand side of the linear system. Has shape (N,) or (N,1).
+
+    Returns
+    -------
+    x : {array, matrix}
+        The converged solution.
+    info : int
+        Provides convergence information:
+          * 0  : successful exit
+          * >0 : convergence to tolerance not achieved, number of iterations
+          * <0 : illegal input or breakdown
+
+    Other parameters
+    ----------------
+    x0 : {array, matrix}
+        Starting guess for the solution (a vector of zeros by default).
+    restart : int, optional
+        Number of iterations between restarts. Larger values increase
+        iteration cost, but may be necessary for convergence.
+        Default is 30.
+    tol : float, optional
+        Tolerances for convergence, ``norm(residual) <= tol*norm(b)``.
+    maxiter : int, optional
+        Maximum number of iterations (restart cycles).  Iteration will stop
+        after maxiter steps even if the specified tolerance has not been
+        achieved.
+    M : {sparse matrix, dense matrix, LinearOperator}
+        Inverse of the preconditioner of A.  M should approximate the
+        inverse of A and be easy to solve for (see Notes).  Effective
+        preconditioning dramatically improves the rate of convergence,
+        which implies that fewer iterations are needed to reach a given
+        error tolerance.  By default, no preconditioner is used.
+    callback : function
+        User-supplied function to call after each iteration.
+    restrt : int, optional
+        DEPRECATED - use `restart` instead.
```

Comment:
We don't need this, just don't pass this argument to the function (or pass/check it to `restart`)
```suggestion
```